### DISCUSSION TITLE
Capturing Llama.cpp stderr in typescript error

### DISCUSSION THREAD

**Discussion by samlhuillier** [withcatai/node-llama-cpp#140] [Q&A] [answered]
What's the best way to capture error messages that get printed to stderr from llama.cpp?
I am using node-llama-cpp in an electron app that let's users load their own gguf files and need more detail on errors than just the javascript stack trace which contains little informative information. For example, I test loaded a corrupted gguf and stderr prints:
```
gguf_init_from_file: invalid magic characters
```
But that doesn't get propagated in typescript. Do you have any tips on how to get the llama.cpp error into typescript?
(FYI I am using the latest beta version - great work on that!)

**Comment by giladgd** [comment]
@samlhuillier There's no straightforward way to get the errors that llama.cpp prints to stderr directly.
I think the best approach for this at the moment may be to run your code in a separate worker process and capture the stderr it prints

**Comment by giladgd** [comment, marked as answer]
@samlhuillier In version `3.0.0-beta.6`, you can now pass a custom logger to the `getLlama` function to catch `llama.cpp`'s error logs there instead of using stderr.
You can see code examples of this version [here](https://github.com/withcatai/node-llama-cpp/pull/105), and the parameter you're looking for is `logger`; the TypeScript types for this parameter should guide you on how to use it.
